sdk/core/core-https/src/xhrHttpsClient.ts

```diff
@@ -59,31 +59,63 @@ export class XhrHttpsClient implements HttpsClient {
     for (const [name, value] of request.headers) {
       xhr.setRequestHeader(name, value);
     }
-    xhr.responseType = request.streamResponseBody ? "blob" : "text";
+    xhr.responseType = request.streamResponseStatusCodes?.size ?? 0 > 0 ? "blob" : "text";
 
     if (isReadableStream(request.body)) {
       throw new Error("Node streams are not supported in browser environment.");
     }
 
     xhr.send(request.body === undefined ? null : request.body);
 
-    if (request.streamResponseBody) {
+    if (request.streamResponseStatusCodes?.size ?? 0 > 0) {
       return new Promise((resolve, reject) => {
         xhr.addEventListener("readystatechange", () => {
           // Resolve as soon as headers are loaded
           if (xhr.readyState === XMLHttpRequest.HEADERS_RECEIVED) {
-            const blobBody = new Promise<Blob>((resolve, reject) => {
+            if (request.streamResponseStatusCodes?.has(xhr.status)) {
+              // eslint-disable-next-line @typescript-eslint/no-shadow
```

Comment:
can avoid needing to disable lint by renaming `resolve` and `reject` here, probably good for readability, even if it's just to call them `blobResolve` and `blobReject`

Reply:
There's another linting rule complaining about the params of Promise constructor callback must be named `resolve` and `reject`.  The old code chose to use `// eslint-disable-next-line @typescript-eslint/no-shadow`.

Reply:
The other error is: "Promise constructor parameters must be named resolve, reject  promise/param-names"

Reply:
I understand now